.yarn/patches/ethereum-cryptography-npm-1.1.2-c16cfd7e8a.patch

```diff
@@ -0,0 +1,17 @@
+diff --git a/utils.js b/utils.js
+index 48dac71e25592c1060017080abd97b6a8790a2ba..4732516f982a9cf6b2708db2f1d823c69c4636cc 100644
+--- a/utils.js
++++ b/utils.js
+@@ -54,10 +54,9 @@ exports.wrapHash = wrapHash;
+ exports.crypto = (() => {
+     const webCrypto = typeof self === "object" && "crypto" in self ? self.crypto : undefined;
+     const nodeRequire = typeof module !== "undefined" &&
+-        typeof module.require === "function" &&
+-        module.require.bind(module);
```

Comment:
@HenryNguyen5 we could use your help with this issue. cc @cds95 

Any idea on the purpose of binding the require fn, `module.require.bind(module)`?

- `ncc` builds are failing with `Error: Cannot find module 'crypto'` at this line. 
- This PR is the temporary fix to avoid this issue.

`ncc` transforms the code to this:

```js
exports.crypto = (() => {
    const webCrypto = typeof self === "object" && "crypto" in self ? self.crypto : undefined;
    const nodeRequire =  true &&
        __nccwpck_require__(84310).bind(module); <-- Note the 
    return {
        node: nodeRequire && !webCrypto ? require("crypto") : undefined,
        web: webCrypto
    };
})();
```

```js
/***/ 84310:
/***/ ((module) => {

function webpackEmptyContext(req) {
	var e = new Error("Cannot find module '" + req + "'");
	e.code = 'MODULE_NOT_FOUND';
	throw e;
}
webpackEmptyContext.keys = () => ([]);
webpackEmptyContext.resolve = webpackEmptyContext;
webpackEmptyContext.id = 84310;
module.exports = webpackEmptyContext;

/***/ }),
```

This leads to the EA crash looping with this error:

```bash
Error: Cannot find module 'crypto'
    at Object.webpackEmptyContext (/Users/christophersastropranoto/Documents/external-adapters-js/packages/sources/layer2-sequencer-health/bundle/index.js:183577:10)
    at /Users/christophersastropranoto/Documents/external-adapters-js/packages/sources/layer2-sequencer-health/bundle/index.js:83270:43
    at Object.78777 (/Users/christophersastropranoto/Documents/external-adapters-js/packages/sources/layer2-sequencer-health/bundle/index.js:83273:3)
    at __nccwpck_require__ (/Users/christophersastropranoto/Documents/external-adapters-js/packages/sources/layer2-sequencer-health/bundle/index.js:183842:43)
    at Object.35681 (/Users/christophersastropranoto/Documents/external-adapters-js/packages/sources/layer2-sequencer-health/bundle/index.js:83194:17)
    at __nccwpck_require__ (/Users/christophersastropranoto/Documents/external-adapters-js/packages/sources/layer2-sequencer-health/bundle/index.js:183842:43)
    at Object.30038 (/Users/christophersastropranoto/Documents/external-adapters-js/packages/sources/layer2-sequencer-health/bundle/index.js:169615:16)
    at __nccwpck_require__ (/Users/christophersastropranoto/Documents/external-adapters-js/packages/sources/layer2-sequencer-health/bundle/index.js:183842:43)
    at Object.55944 (/Users/christophersastropranoto/Documents/external-adapters-js/packages/sources/layer2-sequencer-health/bundle/index.js:168123:14)
    at __nccwpck_require__ (/Users/christophersastropranoto/Documents/external-adapters-js/packages/sources/layer2-sequencer-health/bundle/index.js:183842:43) {
  code: 'MODULE_NOT_FOUND'
}
```

Reply:
I think webpack is trying to bundle `crypto` when it's a native module. You could try the following `ncc` flag: `--external crypto`.

If that doesn't work, I would try upgrading `ncc` to latest to see if it fixes the require statement.

Reply:
I've tried both of those already and still no luck :(

Reply:
I even tried building the `js-ethereum-cryptography` project directly and still see the issue.

Reply:
I’m not sure how important it is to _not_ patch a package.  But you could also abandon ncc and switch it up for something like swc or esbuild + do a prod only install for deps.